src/components/Stepper/Stepper.tsx

```diff
@@ -143,29 +140,63 @@ export const Stepper: FC<StepperProps> = React.forwardRef(
       );
     }, [mergedLocale]);
 
+    const innerSeparatorClassNames: string = mergeClasses([
+      styles.innerSeparator,
+      {
+        [styles.dash]: lineStyle === StepperLineStyle.Dash,
+        [styles.dot]: lineStyle === StepperLineStyle.Dot,
+        [styles.solid]: lineStyle === StepperLineStyle.Solid,
+        [styles.timeline]: variant === StepperVariant.Timeline,
+      },
+    ]);
+
     const stepperClassNames: string = mergeClasses([
       styles.stepper,
       {
+        [styles.timeline]: variant === StepperVariant.Timeline,
         [styles.medium]: size === StepperSize.Medium,
         [styles.small]: size === StepperSize.Small,
         [styles.vertical]: layout === 'vertical',
         [styles.stepperRtl]: htmlDir === 'rtl',
       },
+      classNames,
     ]);
 
-    const activeCircleClassNames: string = mergeClasses([
-      styles.circle,
-      styles.active,
-    ]);
-
-    const visitedCircleClassNames: string = mergeClasses([
-      styles.circle,
-      styles.visited,
+    const stepSizeToIconSizeMap = new Map<StepSize, IconSize>([
+      [StepSize.Large, IconSize.Large],
+      [StepSize.Medium, IconSize.Medium],
+      [StepSize.Small, IconSize.Small],
     ]);
 
-    const circle = (index: number, classes: string): JSX.Element => (
-      <div className={classes}>{index + 1}</div>
-    );
+    const circle = (
+      index: number,
+      classes: string,
+      complete: boolean,
+      size: StepSize,
+      active?: boolean,
+      icon?: IconName
+    ): JSX.Element => {
```

Comment:
I think this JSX.Element is causing issues in typescript implementations where things are expecting React nodes or something along those lines. When i return to that effort I'll sync up with you on it. Just an FYI. no changes requested